cdc/processor.go

```diff
@@ -388,42 +390,43 @@ func (p *processor) localResolvedWorker(ctx context.Context) error {
 }
 
 func (p *processor) updateInfo(ctx context.Context) error {
-	err := p.tsRWriter.WriteInfoIntoStorage(ctx)
+	err := p.tsRWriter.WritePosition(ctx, p.position)
+	if err != nil {
+		return errors.Trace(err)
+	}
 
-	switch errors.Cause(err) {
-	case model.ErrWriteTsConflict:
-		oldInfo, _, err := p.tsRWriter.UpdateInfo(ctx)
+	return retry.Run(func() error {
+		statusChanged, err := p.tsRWriter.UpdateInfo(ctx)
 		if err != nil {
-			return errors.Trace(err)
+			return backoff.Permanent(errors.Trace(err))
 		}
-
+		if !statusChanged {
+			return nil
+		}
+		oldStatus := p.status
 		p.status = p.tsRWriter.GetTaskStatus()
-
 		if p.status.AdminJobType == model.AdminStop || p.status.AdminJobType == model.AdminRemove {
 			err = p.stop(ctx)
 			if err != nil {
-				return errors.Trace(err)
+				return backoff.Permanent(errors.Trace(err))
 			}
-			return errors.Trace(model.ErrAdminStopProcessor)
+			return backoff.Permanent(errors.Trace(model.ErrAdminStopProcessor))
 		}
 
-		p.handleTables(ctx, oldInfo, p.status, oldInfo.CheckPointTs)
+		p.handleTables(ctx, oldStatus, p.status, p.position.CheckPointTs)
 		syncTableNumGauge.WithLabelValues(p.changefeedID, p.captureID).Set(float64(len(p.status.TableInfos)))
 
-		if len(oldInfo.TableInfos) > len(p.status.TableInfos) {
-			// some table is removed, we will not both remove and add table in one operation.
-			// keep CheckpointTs and ResolvedTs as the old in memory cache one.
-			p.status.CheckPointTs = oldInfo.CheckPointTs
-			p.status.ResolvedTs = oldInfo.ResolvedTs
+		err = p.tsRWriter.WriteInfoIntoStorage(ctx)
```

Comment:
Maybe we should move all the code that returns `Permanent` error outside the retried closure.

Reply:
why? I think we should retry when the write operation is conflict

Reply:
I see